fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -213,10 +213,11 @@ struct MetaclusterManagementWorkload : TestWorkload {
 		try {
 			loop {
 				// TODO: check force removal
-				Future<Void> removeFuture = MetaclusterAPI::removeCluster(self->managementDb, clusterName, false);
+				Future<bool> removeFuture = MetaclusterAPI::removeCluster(self->managementDb, clusterName, false);
 				try {
-					Optional<Void> result = wait(timeout(removeFuture, deterministicRandom()->randomInt(1, 30)));
+					Optional<bool> result = wait(timeout(removeFuture, deterministicRandom()->randomInt(1, 30)));
 					if (result.present()) {
+						ASSERT(result.get());
```

Comment:
Based on the implementation of `removeCluster`, I think it's possible to return `false` if the operation times out within the ACTOR, since there is a code path that does not re-throw timeout errors. It looks like it can continue until completion, but the updated flag will stay as false.  I haven't done any runs to see if that's actually possible, but may be worth double-checking

Reply:
The intent is that it should only be possible to get a false value from `removeCluster` if you pass it a timeout duration, which we at the moment aren't doing in line 216. Then if the data cluster removal times out, it won't set the `dataClusterUpdated` field that gets returned upon success of the operation. I think that was the only way that timeouts could be thrown, but if not I can address that.

Reply:
I see, and the wrapper timeout in line 218 returns an empty optional rather than sending a timeout error to the ACTOR, which makes sense. Is the timeout parameter in `removeCluster` actually being exercised or tested anywhere then?

Reply:
Not yet, no. There's a decent way to test this that depends on some changes in the restore PR, so I'm waiting until that to add testing for it.